main.tsx

```diff
@@ -0,0 +1,58 @@
+import { Plugin, MarkdownPostProcessorContext } from 'obsidian';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import BjcpStyleFrontMatter from './src/frontmatter-templates/BjcpStyleFrontMatter';
+import ArticleFrontMatter from 'src/frontmatter-templates/ArticleFrontMatter';
+
+export default class BjcpPlugin extends Plugin {
+  async onload() {
+    this.registerMarkdownPostProcessor((el: HTMLElement, ctx: MarkdownPostProcessorContext) => {
+      console.log('MarkdownPostProcessorContext => ', ctx);
+      const { frontmatter, sourcePath } = ctx;
+
+      if (ctx.getSectionInfo(el).lineStart === 0 && !!frontmatter) {

```

Comment:
This _seems to_ be a better way of figuring out if we're in the "top-most" part of the doc that's being rendered.